flake8_trio.py

```diff
@@ -255,6 +255,135 @@ def check_for_trio100(self, node: Union[ast.With, ast.AsyncWith]) -> None:
                 )
 
 
+# Never have an except Cancelled or except BaseException block with a code path that
+# doesn't re-raise the error
+class Visitor103_104(ast.NodeVisitor):
+    def __init__(self) -> None:
+        super().__init__()
+        self.problems: List[Error] = []
```

Comment:
Tempted to have a `Flake8TrioVisitor` superclass which sets up self.problems, and then we can iterate over it's `__subclasses__()` (iirc) in the Plugin.

Reply:
`__subclasses__()` works!

and could further clean up `Plugin.run` with a `classmethod`
```python
class TrioVisitor(ast.NodeVisitor):
    def __init__(self) -> None:
        super().__init__()
        self.problems: List[Error] = []

    @classmethod
    def run(cls, tree: ast.AST) -> Generator[Error, None, None]:
        visitor = cls()
        visitor.visit(tree)
        yield from visitor.problems
```
```python
def run(self) -> [...]:
    for v in TrioVisitor.__subclasses__():
        yield from v.run(self._tree)
```